src/riak_kv_qry_buffers_ldb.erl

```diff
@@ -0,0 +1,172 @@
+%%-------------------------------------------------------------------
+%%
+%% riak_kv_qry_buffers_ldb: Riak SQL query result disk-based temp storage
+%%                          (leveldb backend ops)
+%%
+%% Copyright (C) 2016 Basho Technologies, Inc. All rights reserved
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%%-------------------------------------------------------------------
+
+%% @doc leveldb operations for Riak TS query buffers
+
+-module(riak_kv_qry_buffers_ldb).
+
+-export([new_table/2,
+         delete_table/3,
+         add_rows/5,
+         fetch_rows/3]).
+
+
+%% leveldb instance parameters
+-define(LDB_WRITE_BUFFER_SIZE, 10*1024*1024).  %% 10 M should be enough for everybody
+
+
+-type errors() :: ldb_put_failed.
+-export_type([errors/0]).
+
+-spec new_table(binary(), string()) -> {ok, eleveldb:db_ref()} | {error, term()}.
+new_table(Table, Root) ->
+    Path = filename:join(Root, binary_to_list(Table)),
+    _ = os:cmd(fmt("mkdir -p '~s'", [Path])),
```

Comment:
We now have our very own `riak_kv_ts_util:rm_rf/1` (916c59e). For the mkdir case, there is `filelib:ensure_dir/1`.